plan/planbuilder.go

```diff
@@ -826,13 +826,38 @@ func buildShowEventsSchema() *expression.Schema {
 	return schema
 }
 
-// getShowColNamesAndTypes gets column names and types. If the `ftypes` is empty, every column is set to varchar type.
```

Comment:
This function is only used by `buildShowSchema`, I think
`buildShowSchemaWithNamesAndTypes` is better.